libeatmydata/libeatmydata.c

```diff
@@ -236,3 +246,30 @@ int sync_file_range(int fd, off64_t offset, off64_t nbytes, unsigned int flags)
 	return (libc_sync_file_range)(fd, offset, nbytes, flags);
 }
 #endif
+
+#if defined(F_FULLFSYNC) && defined(__APPLE__)
+
+/* fcntl does not have a va_list version. We have to trust this one seen in
+public Darwin sources:
+int __FCNTL(int, int, void *); */
+
+int LIBEATMYDATA_API fcntl(int fd, int cmd, ...)
+{
+	if ((eatmydata_is_hungry() && (cmd == F_FULLFSYNC))) {
+		pthread_testcancel();
```

Comment:
This one might be redundant given that we call actual fcntl on the next line.

Reply:
Probably is, I'd expect that if it's a cancellation point that fcntl would take care of it for other fctnl commands.

Reply:
anyway, i can fix that in merging.